### ISSUE TITLE
AFX 1.4.9 - NullPointerException generated when loading fontawesome-webfont.woff

### ISSUE THREAD

**Issue by wfouche** [asciidocfx/AsciidocFX#246]
A document that I could open and display in all previous versions of AsciidocFX, now generates the following exception when opened with the latest release of AsciidocFX:

````
ERROR => UT005023: Exception handling request to /afx/fonts/fontawesome-webfont.woff
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.lang.NullPointerException

	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:982)

	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)

	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)

	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)

	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)

	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)

	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)

	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)

	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)

	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)

	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)

	at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:87)

	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)

	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)

	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)

	at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:77)

	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)

	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)

	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)

	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)

	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)

	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)

	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)

	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)

	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)

	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)

	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)

	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)

	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)

	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)

	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)

	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)

	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)

	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)

	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)

	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)

	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:285)

	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:264)

	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)

	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:175)

	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:202)

	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:802)

	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)

	at java.lang.Thread.run(Unknown Source)

Caused by: java.lang.NullPointerException: null

	at java.lang.String.contains(Unknown Source)

	at com.kodcu.controller.Payload.getFinalURI(Payload.java:53)

	at com.kodcu.controller.CommonResource.processPayload(CommonResource.java:40)

	at com.kodcu.controller.AllController.all(AllController.java:37)

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)

	at java.lang.reflect.Method.invoke(Unknown Source)

	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221)

	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:136)

	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:114)

	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)

	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)

	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)

	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963)

	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897)

	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)

	... 44 common frames omitted
````

**Comment by rahmanusta** [comment]
Hi @wfouche how do you load fontawesome-webfont.woff in document? Could you please give a sample to test it? Thanks for reporting

**Comment by wfouche** [comment]
It is very easy to reproduce this issue. I maintain a task list in AsciidocFX.

````
= Planning

== Project 1

Task List::

* Task1
* Task2
* Task3
````
![clipboard02](https://cloud.githubusercontent.com/assets/13682850/18380310/fe62405c-7676-11e6-9ad5-cd83d86d4d52.jpg)

The exception occurs when I mark  a task as done with [x].
````
= Planning

== Project 1

Task List::

* Task1
* [x] Task2
* Task3
````
![clipboard01](https://cloud.githubusercontent.com/assets/13682850/18380394/6b3ddf88-7677-11e6-9acd-ca5d7f8acb6d.jpg)

**Comment by rahmanusta** [comment]
Thank you, I reproduced the issue. I will focus on that.

**Comment by wfouche** [comment]
@rahmanusta - the test document is rendered correctly in AsciidocFX 1.4.6, but it not working in AsciidocFX 1.4.7 and beyond.

**Comment by rahmanusta** [comment]
Hey @wfouche, can you try latest version 1.5.0?

**Comment by wfouche** [comment]
Thanks @rahmanusta !!

This issue is resolved in AsciifocFX 1.5.0.

**Comment by rahmanusta** [comment]
Thank you @wfouche :+1:
